src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -421,30 +421,48 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     } getOrElse Future.successful(())
   }
 
-  private[service] def validateBucketObjectUri(userEmail: WorkbenchEmail, googleProject: GoogleProject, gcsUriOpt: Option[GcsPath])(implicit executionContext: ExecutionContext): Future[Unit] = {
 
+  private def validateClusterRequestBucketObjectUri(userEmail: WorkbenchEmail, googleProject: GoogleProject, clusterRequest: ClusterRequest)(implicit executionContext: ExecutionContext): Future[Unit] = {
+    for {
+      _ <- clusterRequest.jupyterUserScriptUri match {
+        case Some(userScriptUri) => validateBucketObjectUri(userEmail, googleProject, userScriptUri.toUri)
+        case None => Future.successful(())
+      }
+
+      _ <- clusterRequest.userJupyterExtensionConfig match {
+        case Some(config) =>
+          val extensionsToValidate = (config.nbExtensions.values ++ config.serverExtensions.values ++ config.combinedExtensions.values).filter(_.startsWith("gs://"))
+          Future.traverse(extensionsToValidate)(x => validateBucketObjectUri(userEmail, googleProject, x))
+        case None => Future.successful(())
+      }
+    } yield()
+  }
+  private[service] def validateBucketObjectUri(userEmail: WorkbenchEmail, googleProject: GoogleProject, gcsUri: String)(implicit executionContext: ExecutionContext): Future[Unit] = {
+
+    val gcsUriOpt = parseGcsPath(gcsUri)
     gcsUriOpt match {
-      case None => Future.successful(())
-      case Some(gcsPath) =>
+      case Right(gcsPath) =>
         if (gcsPath.toUri.length > bucketPathMaxLength)
-          throw BucketObjectException(gcsPath)
+          throw BucketObjectException(gcsUri)
```

Comment:
Since this returns a Future, should probably do `Future.failed(BucketObjectException(gcsUri))` instead of throwing exceptions directly.